clients/admin-ui/src/features/dataset/EditDatasetForm.tsx

```diff
@@ -0,0 +1,91 @@
+import { Box, Button, Stack } from "@fidesui/react";
+import { Form, Formik } from "formik";
+import { useState } from "react";
+import { useSelector } from "react-redux";
+
+import { selectDataCategories } from "~/features/taxonomy/data-categories.slice";
+
+import { CustomSelect, CustomTextInput } from "../common/form/inputs";
+import { DATA_QUALIFIERS } from "./constants";
+import DataCategoryInput from "./DataCategoryInput";
+import { Dataset } from "./types";
+
+type FormValues = Partial<Dataset>;
+
+interface Props {
+  values: FormValues;
+  onClose: () => void;
+  onSubmit: (values: FormValues) => void;
+}
+
+const EditDatasetForm = ({ values, onClose, onSubmit }: Props) => {
+  const initialValues: FormValues = {
+    name: values.name ?? "",
+    description: values.description ?? "",
+    retention: values.retention ?? "",
+    data_qualifier: values.data_qualifier,
```

Comment:
Is this value and the values below always guaranteed?

Reply:
Yup, `data_qualifier` is a required field in the db. `data_categories` could be undefined. we could do a `values.data_qualifier ?? []` here if we wanted, though it's not necessary (it's more necessary for strings since IIRC formik doesn't like undefined string fields)